src/static/helpers/slasHelper.ts

```diff
@@ -134,7 +136,7 @@ export async function authorize(
     parameters: {
       client_id: slasClient.clientConfig.parameters.clientId,
       channel_id: slasClient.clientConfig.parameters.siteId,
-      code_challenge: codeChallenge,
+      ...(!privateClient && {code_challenge: codeChallenge}),
```

Comment:
Generating the code challenge isn't a trivial operation, especially on old mobile devices. Instead of not using the code challenge that was already created when in private client. You should not generate it in the first place. 

Maybe something along the line of.

```
let clientOptions = {} 
if (privateClient) {
   clientOptions.code_challenge = await generateCodeChallenge(codeVerifier)
}
```